crates/pop-cli/src/commands/hash.rs

```diff
@@ -0,0 +1,444 @@
+// SPDX-License-Identifier: GPL-3.0
+
+use self::Command::*;
+use super::*;
+use anyhow::{anyhow, Result};
+use clap::{
+	builder::{PossibleValue, PossibleValuesParser, TypedValueParser},
+	Arg, Args, Error,
+};
+use sp_core::{
+	bytes::{from_hex, to_hex},
+	hashing::*,
+};
+use std::{ffi::OsStr, ops::Deref, str::FromStr};
+use strum_macros::Display;
+
+const CONCAT: &'static str = "Whether to append the source data to the hash.";
+const DATA: &'static str =
+	"The data to be hashed: input directly or specified as a path to a file.";
+const LENGTH: &'static str = "The length of the resulting hash, in bits.";
+const MAX_CODE_SIZE: u64 = 3 * 1024 * 1024;
+
+/// Arguments for hashing.
+#[derive(Args)]
+#[command(args_conflicts_with_subcommands = true)]
+pub(crate) struct HashArgs {
+	/// Hash data using a supported hash algorithm.
+	#[command(subcommand)]
+	pub(crate) command: Command,
+}
+
+/// Hash data using a supported hash algorithm.
+#[derive(Subcommand)]
+pub(crate) enum Command {
+	/// Hashes data using the BLAKE2b cryptographic hash algorithm.
+	#[clap(alias = "b2")]
+	Blake2 {
+		#[arg(help = LENGTH, value_parser = SupportedLengths::new([64, 128, 256, 512]))]
+		length: u16,
+		#[arg(help = DATA)]
+		data: Data,
+		#[arg(short, help = CONCAT, long)]
+		concat: bool,
+	},
+	/// Hashes data using the Keccak cryptographic hash algorithm.
+	#[clap(alias = "kk")]
+	Keccak {
+		#[arg(help = LENGTH, value_parser = SupportedLengths::new([256, 512]))]
+		length: u16,
+		#[arg(help = DATA)]
+		data: Data,
+	},
+	/// Hashes data using the SHA-2 cryptographic hash algorithm.
+	#[clap(alias = "s2")]
+	Sha2 {
+		#[arg(help = LENGTH, value_parser = SupportedLengths::new([256]))]
+		length: u16,
+		#[arg(help = DATA)]
+		data: Data,
+	},
+	/// Hashes data using the non-cryptographic xxHash hash algorithm.
+	#[clap(alias = "xx", name = "twox")]
+	TwoX {
+		#[arg(help = LENGTH, value_parser = SupportedLengths::new([64, 128, 256]))]
+		length: u16,
+		#[arg(help = DATA)]
+		data: Data,
+		#[arg(short, help = CONCAT, long)]
+		concat: bool,
+	},
+}
+
+impl Command {
+	/// Executes the command.
+	pub(crate) fn execute(&self) -> Result<()> {
+		let hash = self.hash()?;
+		let additional_info = format!("(Source: {}, Output: {} bytes)", self.data(), hash.len());
+		let output =
+			format!("{} {}", to_hex(&self.hash()?, false), console::style(additional_info).dim());
+		println!("{output}");
```

Comment:
For consistency can we use `cli.info` or `cli.success` instead?

Reply:
Pushed in 18b1408d